xgboost_ray/main.py

```diff
@@ -512,18 +517,51 @@ def _shutdown(actors: List[ActorHandle],
         queue.shutdown()
     if event:
         event.shutdown()
+    if placement_group:
+        remove_placement_group(placement_group)
+
+def _create_communication_processes():
+    # Create Queue and Event actors and make sure to colocate with driver node.
+    node_ip = ray.services.get_node_ip_address()
+    # Have to explicitly set num_cpus to 0.
+    placement_option = {"num_cpus": 0, "resources": {f"node:{node_ip}": 0.01}}
+    queue = Queue(actor_options=placement_option)  # Queue actor
+    stop_event = Event(actor_options=placement_option)  # Stop event actor
+    return queue, stop_event
+
+def _create_placement_group(cpus_per_actor, gpus_per_actor,
+                            resources_per_actor, num_actors):
+    resources_per_bundle = {"CPU": cpus_per_actor, "GPU": gpus_per_actor}
```

Comment:
Can you add a check that cpu per actor & gpu per actors are not both 0? Placement group won't work in that case

Reply:
Added!